core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/src/org/polarsys/capella/core/platform/sirius/ui/handlers/DeleteHandler.java

```diff
@@ -36,10 +39,7 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
 
     if (selection instanceof IStructuredSelection) {
       List<?> elementsToRemove = ((IStructuredSelection) selection).toList();
-      List<EObject> eObjectsToRemove = elementsToRemove.stream()//
-          .filter(EObject.class::isInstance) //
-          .map(EObject.class::cast).collect(Collectors.toList());
-
+      List<EObject> eObjectsToRemove = getEObjectsToRemove(elementsToRemove);
       if (!eObjectsToRemove.isEmpty()) {
 
         ExecutionManager executionManager = TransactionHelper.getExecutionManager(eObjectsToRemove);
```

Comment:
**It should be eObjectsToRemove!**

 new CapellaDeleteCommand(executionManager, ~~elementsToRemove~~ eObjectsToRemove, true,
withConfirmDeletion(), true)`